### PR TITLE
Tests: fix mock.patch order

### PR DESCRIPTION
**Pull request by stsewd**
They should be bottom up.

### CODE REVIEW

**Comment by stsewd** [comment]
okay, so the error was for two reasons:

- we were returning a mock objects, when we were expecting a boolean
- we were returning an old value from the api, since the mocked data was from the time the mock was defined,
  so changed this to be a function instead, so we always return the current values.

File: readthedocs/projects/tests/mockers.py [hunk 1]
```diff
@@ -238,7 +237,7 @@ def _mock_api(self):
 
         self.requestsmock.get(
             f'{settings.SLUMBER_API_HOST}/api/v2/project/{self.project.pk}/active_versions/',
-            json={
+            json=lambda request, context: {
```
[the remark this answers is not in the document]
Comment (humitos) [reply readthedocs/projects/tests/mockers.py:240]: This comment applies for the other places where a lambda function was used. I don't see any obvious case where we need this to be fresh values. Can you expand on this, please?

File: readthedocs/projects/tests/test_build_tasks.py [hunk 2]
```diff
@@ -146,14 +148,15 @@ def test_build_sphinx_formats(self, load_yaml_config, formats, builders):
...
-    @mock.patch('readthedocs.projects.tasks.builds.UpdateDocsTask.build_docs_class')
     @mock.patch('readthedocs.projects.tasks.builds.load_yaml_config')
+    @mock.patch('readthedocs.projects.tasks.builds.UpdateDocsTask.build_docs_class')
+    @mock.patch('readthedocs.projects.tasks.builds.UpdateDocsTask.build_docs_html')
     def test_build_formats_only_html_for_external_versions(self, build_docs_html, build_docs_class, load_yaml_config):
         load_yaml_config.return_value = self._config_file({
             'version': 2,
             'formats': 'all',
         })
+        build_docs_html.return_value = True
```
Comment (humitos) [inline readthedocs/projects/tests/test_build_tasks.py:159]: Why this mock requires a return value, but the following one (`test_build_respects_formats_mkdocs`, -which is almost identical) doesn't?

**Review by humitos — COMMENTED** [review]
I'm not 100% convinced this PR is accurate or, at least, I don't fully understand it. Also, it seems that tests are passing in master so this PR is not required? 🤔

File: readthedocs/projects/tests/test_build_tasks.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/projects/tests/test_build_tasks.py]: Looks like isn't required anymore after fixing the api mocks.

**Review by humitos — APPROVED** [review]
I hit this issue today and arrived at the same conclusion than you 👍🏼

**Pull request merged**